lib/addons/kube-proxy/index.ts

```diff
@@ -1,15 +1,54 @@
+import { CfnAddon, KubernetesVersion } from "aws-cdk-lib/aws-eks";
+import { Construct } from "constructs";
+import { ClusterInfo } from "../../spi";
+import { logger } from "../../utils";
 import { CoreAddOn } from "../core-addon";
 
+const versionMap: Map<KubernetesVersion, string> = new Map([
+    [KubernetesVersion.of("1.27"), "v1.27.1-eksbuild.1"],
+    [KubernetesVersion.V1_26, "v1.26.2-eksbuild.1"],
+    [KubernetesVersion.V1_25, "v1.25.6-eksbuild.1"],
+    [KubernetesVersion.V1_24, "v1.24.7-eksbuild.2"],
+    [KubernetesVersion.V1_23, "v1.23.7-eksbuild.1"],
+]);
+
 /**
  * Implementation of KubeProxy EKS add-on.
  */
 export class KubeProxyAddOn extends CoreAddOn {
 
+    version: string;
+
     constructor(version?: string) {
         super({
             addOnName: "kube-proxy",
-            version: version ?? "v1.25.6-eksbuild.1",
+            version: version ?? "auto",
             saName: "kube-proxy"
         });
+        this.version = version ?? "auto";
+    }
+
+    deploy(clusterInfo: ClusterInfo): Promise<Construct> {
```

Comment:
Hey Zach, while this implementation works, it almost nullifies the inheritance  - there is very little reason to derive this from a core addon. 

I think we what we should do in this case is to enable this type of version selection for any core addon. 

That means that the core addon should be able to accept optionally something like a versionProviderFunction that takes cluster info and spits out the version. 

The kubeproxy will supply its function that will use the map. However, the default function will to just return whatever customers passed. 

In this way, other core addons, the number of which will grow, will be able to address this aspect without sacrificing the central implementation component of such addons.

Reply:
Yep, that approach makes a ton of sense.  We should discuss implementation.